www/scipylib/building/windows.rst

```diff
@@ -5,374 +5,213 @@ Building From Source on Windows
 #####
 
-The easiest way to install MinGW is to use the MinGW installer from here. The
-current (as of 3 August, 06) candidate distribution with gcc and g77 3.4.5 is
-reported to work best. After installation you need to put the MinGW\bin
-directory on the path.
+The Mingw-w64_ project provides Windows versions of the free GNU compilers **gcc** and
+**gfortran**. These are the compilers most NumPy SciPy developers work with and hence
+the compilers that are supported the best by build scripts in both libraries. Also,
+as indicated in the name, they also form the basis of the ongoing Mingwpy_ project
+mentioned previously. Thus, from a long-term perspective, these compilers may be
+the optimal ones to use. Installation instructions can be found `here <http://mingw-w64.org/doku.php/download>`__.
+
+.. _Mingw-w64: http://mingw-w64.org/doku.php/
 
 Cygwin
 ######
 
-Cygwin can be conveniently installed and updated with their 
-`convenient installer <http://www.cygwin.com/setup.exe>`.
-
-Make sure that the ``gcc`` and ``gcc-mingw`` packages in the "Devel" section
-are selected. If you want to use the official Python distribution (recommended)
-and don't want to get confused, do not select the Python option in the Cygwin
-installer. The Cygwin tools should be used from within the Cygwin ``bash`` 
-shell available from the start menu after installation.  See this  
-`tutorial <http://cplus.about.com/od/compilersandides/l/aa061304a.htm>`_
-explaining the basics of Cygwin.
-
-Visual Studio
-#############
-
-The installation is quite straightforward; see the documentation provided by
-Microsoft for additional help. The command line tools are most conveniently
-used from the Visual Studio command prompt available from  the start menu.
+A POSIX compatible Linux-like environment for Windows, Cygwin_ is a very useful tool,
+as it allows to compile and use many Unix tools without modification. It can also be
+used to build libraries ATLAS_, which at the moment is very Unix-oriented, although
+that may be subject to change as we will discuss later on. Installation instructions
+for Cygwin_ can be found `here <https://cygwin.com/install.html>`__. When using the
+installer (either 32-bit or 64-bit depending on your computer), **make sure to search
+for and select** packages with the keyword **gcc** in them. The installer may also miss
+several important DLL's as pointed out `here <http://stackoverflow.com/questions/32897685/cannot-compile-anything-with-gcc-on-cygwin32-missing-cygisl-10-dll>`__,
+so double check that you have them marked as well during installation. Note that
+even if you forget to install a package, you can always run the installer again to
+install additional ones.
+
+In addition, Cygwin also offers its own **identical** packages for Mingw-w64_ that you can
+install by searching from **mingw64** in the packages and then selecting those that contain
+**i686** if you're using 32-bit or **x86_64** if you're using 64-bit. If you choose this option,
```

Comment:
It's probably worth describing the difference between cygwin's gcc, which creates applications that link against the (GPL) cygwin dll and can only run with cygwin present, vs the cygwin hosted mingw64 cross compilers that run within cygwin but build native binaries that do not themselves require cygwin to run.